doc/source/ray-core/actors.rst

```diff
@@ -66,373 +42,189 @@ Creating an actor
-      Counter* CreateCounter() {
+      static Counter *CreateCounter() {
           return new Counter();
-        }
-
-      RAY_REMOTE(&Counter::Increment, &Counter::GetCounter,
-                 &Counter::Reset, CreateCounter);
-
-      // Create an actor with a factory method.
-      ray::Actor(CreateCounter).Remote();
-
-When the above actor is instantiated, the following events happen.
-
-1. A node in the cluster is chosen and a worker process is created on that node
-   for the purpose of running methods called on the actor.
-2. A ``Counter`` object is created on that worker and the ``Counter``
-   constructor is run.
-
-Actor Methods
--------------
-
-Methods of the actor can be called remotely.
-
-.. tabbed:: Python
-
-    .. code-block:: python
-
-        counter_actor = Counter.remote()
-
-        assert ray.get(counter_actor.increment.remote()) == 1
-
-        @ray.remote
-        class Foo(object):
-
-            # Any method of the actor can return multiple object refs.
-            @ray.method(num_returns=2)
-            def bar(self):
-                return 1, 2
-
-        f = Foo.remote()
-
-        obj_ref1, obj_ref2 = f.bar.remote()
-        assert ray.get(obj_ref1) == 1
-        assert ray.get(obj_ref2) == 2
-
-.. tabbed:: Java
-
-    .. code-block:: java
-
-        ActorHandle<Counter> counterActor = Ray.actor(Counter::new).remote();
-        // Call an actor method with a return value
-        Assert.assertEquals((int) counterActor.task(Counter::increment).remote().get(), 1);
-        // Call an actor method without return value. In this case, the return type of `remote()` is void.
-        counterActor.task(Counter::reset, 10).remote();
-        Assert.assertEquals((int) counterActor.task(Counter::increment).remote().get(), 11);
+      };
 
-.. tabbed:: C++
+      RAY_REMOTE(&Counter::Increment, CreateCounter);
 
-    .. code-block:: c++
+      // Create an actor from this class.
+      // `ray::Actor` takes a factory method that can produce
+      // a `Counter` object. Here, we pass `Counter`'s factory function
+      // as the argument.
+      auto counter = ray::Actor(CreateCounter).Remote();
 
-        ray::ActorHandle<Counter> counter_actor = ray::Actor(CreateCounter).Remote();
-        // Call an actor method with a return value
-        assert(*counter_actor.Task(&Counter::Increment).Remote().Get(), 1);
-        // Call an actor method without return value. In this case, the return type of `Remote()` is void.
-        counter_actor.Task(&Counter::Reset).Remote(10);
-        assert(*counter_actor.Task(&Counter::Increment).Remote().Get(), 11);
+Specifying required resources
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. _actor-resource-guide:
 
-Specifying Resources
---------------------
-
-You can specify that an actor requires CPUs or GPUs in the decorator. While Ray has built-in support for CPUs and GPUs, Ray can also handle custom resources.
+You can specify resource requirements in actors too (see the `Actors section
```

Comment:
Maybe copy and/or link the explanation from the tasks walkthrough? I don't think we should assume that people have read the tasks one yet.